src/main/java/org/opensearch/knn/plugin/BlockedModelIds.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.plugin;
+
+import org.apache.logging.log4j.LogManager;
+import org.opensearch.Version;
+import org.opensearch.cluster.Diff;
+import org.opensearch.cluster.NamedDiff;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BlockedModelIds implements Metadata.Custom {
+
+    public static Logger logger = LogManager.getLogger(BlockedModelIds.class);
+    public static final String TYPE = "opensearch-knn-blocked-models";
+
+    List<String> blockedModelIds;
+
+    public BlockedModelIds(List<String> blockedModelIds) {
+        this.blockedModelIds = blockedModelIds;
+    }
+
+    @Override
+    public EnumSet<Metadata.XContentContext> context() {
+        return Metadata.ALL_CONTEXTS;
+    }
+
+    @Override
+    public Diff<Metadata.Custom> diff(Metadata.Custom custom) {
+        return null;
+    }
+
+    @Override
+    public String getWriteableName() {
+        return TYPE;
+    }
+
+    @Override
+    public Version getMinimalSupportedVersion() {
+        return Version.CURRENT.minimumCompatibilityVersion();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeVInt(blockedModelIds.size());
+        for (String modelId : blockedModelIds) {
+            out.writeString(modelId);
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        return null;
+    }
+
+    public BlockedModelIds(StreamInput in) throws IOException {
+        int size = in.readVInt();
+        List<String> modelIds = new ArrayList<>(size);
+        for (int i = 0; i < size; i++) {
+            String modelId = in.readString();
+            modelIds.add(modelId);
+        }
+        this.blockedModelIds = modelIds;
+    }
+
+    public List<String> getBlockedModelIds() {
+        return blockedModelIds;
+    }
+
+    public void remove(String modelId) {
+        if (blockedModelIds.contains(modelId)) {
```

Comment:
we can directly call remove i believe , if it doesn't exist it will return false